### PR TITLE
Feature processing data in models

### PR DESCRIPTION
**Pull request by Graciano1997**
In this Milestone I did the following :

Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).

- Set up associations between models.

[Use models to insert data](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#use-models-to-insert-data)

Familiarized myself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
Opened rails console for your project.

[Created custom methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#create-custom-methods)

-  My models include some additional methods.

- Users
A method that returns the 3 most recent posts for a given user.
- Posts
A method that updates the posts counter for a user.
A method that returns the 5 most recent comments for a given post.

- Comments
A method that updates the comments counter for a post.

- Likes
A method that updates the likes counter for a post.

I checked all the methods in rails c

Fix linter issues
Update readme

### CODE REVIEW

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Graciano1997, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  belongs_to :post, foreign_key: 'post_id'
+
+  def like_counter_update(post)
+    post_object = Post.where(id: post.id)[0]
+    if post_object.likes_counter.nil?
+      Post.where(id: post.id).update(likes_counter: 1)
+    else
+      Post.where(id: post.id).update(likes_counter: post_object.likes_counter + 1)
+    end
+  end
```
[the remark this answers is not in the document]
Comment (Graciano1997) [reply app/models/like.rb]: @Qoosim, I want to thank you for your Review 🙏 🙏

File: app/models/user.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'user_id'
 
-  def show_post_user(user)
-    Comment.where(user_id: user.id).last(3)
+  def user_recent_posts(user)
+    Post.where(author_id: user.id).order('created_at Desc').limit(3)
   end
```
Comment (Strangeal) [inline app/models/user.rb]: - [ ] Great job implementing the changes suggested by the previous reviewer. However, The `user_recent_posts` method still needs some adjustments. This method is to return the 3 most recent posts created in the database. I can see you're returning the 3 most recent posts but you're returning the post based on the author_id which shouldn't be so. 
    - If you look at the previous reviewer's comment your would see he provided a clear example of how to achieve this.

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @Graciano1997, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text

 ## Highlights
- [ ] Used `Rails Active Callbacks` in calling private methods ✅


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Omar-Muhamad — APPROVED** [review]
# Approved ✔
### Hi @Graciano1997,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**